### YamlMime:ManagedReference
items:
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs
  id: DataGridViewDataErrorEventArgs
  children:
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  langs:
  - csharp
  name: DataGridViewDataErrorEventArgs
  nameWithType: DataGridViewDataErrorEventArgs
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs
  type: Class
  summary: "提供資料給<xref href=&quot;System.Windows.Forms.DataGridView.DataError&quot;></xref>事件。"
  remarks: "處理<xref:System.Windows.Forms.DataGridView.DataError>事件可讓您控制代碼控制項 （例如，藉由外部資料來源） 之外的程式碼擲回的例外狀況。</xref:System.Windows.Forms.DataGridView.DataError> 使用<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A>屬性來判斷狀態的<xref:System.Windows.Forms.DataGridView>時的例外狀況。</xref:System.Windows.Forms.DataGridView> </xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A> 使用<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A>屬性，以擷取例外狀況資料。</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> 如果您想要處理的例外狀況，由其他事件處理常式，設定<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A>屬性`true`。</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A>       <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A>和<xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A>屬性通常會指出發生資料錯誤的資料格。</xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A> </xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A> 發生錯誤時在外部資料來源，不過，資料來源可能不會提供發生錯誤的資料行。 在此情況下，<xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A>屬性通常表示在發生錯誤時的目前儲存格的資料行。</xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A>"
  example:
  - "The following code example demonstrates how to respond to information provided by the DataGridViewDataErrorEventArgs class. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/t-system.windows.forms.d_56_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/t-system.windows.forms.d_56_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/t-system.windows.forms.d_56_1.vb)]"
  syntax:
    content: 'public class DataGridViewDataErrorEventArgs : System.Windows.Forms.DataGridViewCellCancelEventArgs'
  inheritance:
  - System.ComponentModel.CancelEventArgs
  - System.Windows.Forms.DataGridViewCellCancelEventArgs
  implements: []
  inheritedMembers:
  - System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex
  - System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)
  id: '#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)'
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  type: Constructor
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "初始化的新執行個體<xref href=&quot;System.Windows.Forms.DataGridViewDataErrorEventArgs&quot;></xref>類別。"
  syntax:
    content: public DataGridViewDataErrorEventArgs (Exception exception, int columnIndex, int rowIndex, System.Windows.Forms.DataGridViewDataErrorContexts context);
    parameters:
    - id: exception
      type: System.Exception
      description: "發生的例外狀況。"
    - id: columnIndex
      type: System.Int32
      description: "引發的儲存格的資料行索引<xref href=&quot;System.Windows.Forms.DataGridView.DataError&quot;> </xref>。"
    - id: rowIndex
      type: System.Int32
      description: "引發的儲存格的資料列索引<xref href=&quot;System.Windows.Forms.DataGridView.DataError&quot;> </xref>。"
    - id: context
      type: System.Windows.Forms.DataGridViewDataErrorContexts
      description: "位元組合<xref href=&quot;System.Windows.Forms.DataGridViewDataErrorContexts&quot;></xref>值，指出錯誤發生時的內容。"
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  id: Context
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: Context
  nameWithType: DataGridViewDataErrorEventArgs.Context
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  type: Property
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "取得狀態的相關詳細資料<xref href=&quot;System.Windows.Forms.DataGridView&quot;></xref>錯誤發生時。"
  remarks: "成員<xref:System.Windows.Forms.DataGridViewDataErrorContexts>列舉型別可能會使用位元結合`OR`運算子來呈現的資料繫結狀態<xref:System.Windows.Forms.DataGridView>資料錯誤發生時。</xref:System.Windows.Forms.DataGridView> </xref:System.Windows.Forms.DataGridViewDataErrorContexts> 例如，如果使用者輸入無效的資料格的值 （例如，輸入中需要日期的儲存格的名稱），然後選取不同的資料格，<xref:System.Windows.Forms.DataGridView>會嘗試將無效的資料格的值認可。</xref:System.Windows.Forms.DataGridView> 當認可失敗時，<xref:System.Windows.Forms.DataGridView>會引發<xref:System.Windows.Forms.DataGridView.DataError>事件，其內容屬性會具有<xref:System.Windows.Forms.DataGridViewDataErrorContexts>和<xref:System.Windows.Forms.DataGridViewDataErrorContexts>。</xref:System.Windows.Forms.DataGridViewDataErrorContexts></xref:System.Windows.Forms.DataGridViewDataErrorContexts>值</xref:System.Windows.Forms.DataGridView.DataError></xref:System.Windows.Forms.DataGridView>"
  example:
  - "The following code example demonstrates how to investigate the error context. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_134_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_134_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_134_1.vb)]"
  syntax:
    content: public System.Windows.Forms.DataGridViewDataErrorContexts Context { get; }
    return:
      type: System.Windows.Forms.DataGridViewDataErrorContexts
      description: "位元組合<xref href=&quot;System.Windows.Forms.DataGridViewDataErrorContexts&quot;></xref>值，指定發生錯誤的內容。"
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context*
  exceptions: []
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  id: Exception
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: Exception
  nameWithType: DataGridViewDataErrorEventArgs.Exception
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  type: Property
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "取得表示錯誤的例外狀況。"
  remarks: "若要處理此例外狀況由多個<xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>，藉由設定擲回的例外狀況，一次的處理常式結尾<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A>屬性`true`。</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A> </xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>"
  example:
  - "The following code example demonstrates checking the Exception property. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_249_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_249_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_249_1.vb)]"
  syntax:
    content: public Exception Exception { get; }
    return:
      type: System.Exception
      description: "<xref:System.Exception>表示錯誤。</xref:System.Exception>"
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*
  exceptions: []
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  id: ThrowException
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: ThrowException
  nameWithType: DataGridViewDataErrorEventArgs.ThrowException
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  type: Property
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "取得或設定值，指出是否要擲回的例外狀況之後<xref href=&quot;System.Windows.Forms.DataGridViewDataErrorEventHandler&quot;></xref>委派是它已完成。"
  remarks: "將此屬性設定為`false`如果<xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>所代表的例外狀況有處理<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A>屬性，而且不想要傳播至另一個例外狀況<xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>，可能會處理<xref:System.Windows.Forms.DataGridView.DataError>事件。</xref:System.Windows.Forms.DataGridView.DataError> </xref:System.Windows.Forms.DataGridViewDataErrorEventHandler> </xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> </xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>       偵錯，它可用來傳播例外狀況，並檢查錯誤的相關內容資訊的堆疊追蹤。"
  example:
  - "The following code example demonstrates using the ThrowException property to indicate that the exception should not be thrown. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_202_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_202_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_202_1.vb)]"
  syntax:
    content: public bool ThrowException { get; set; }
    return:
      type: System.Boolean
      description: "<xref uid=&quot;langword_csharp_true&quot; name=&quot;true&quot; href=&quot;&quot;></xref>如果應該擲回例外狀況。否則， <xref uid=&quot;langword_csharp_false&quot; name=&quot;false&quot; href=&quot;&quot;> </xref>。 預設值是<xref uid=&quot;langword_csharp_false&quot; name=&quot;false&quot; href=&quot;&quot;> </xref>。"
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException*
  exceptions:
  - type: System.ArgumentException
    commentId: T:System.ArgumentException
    description: "若要設定這個屬性時<xref uid=&quot;langword_csharp_true&quot; name=&quot;true&quot; href=&quot;&quot;> </xref>、<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*>屬性值是<xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>。</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*>"
  platform:
  - net462
references:
- uid: System.Windows.Forms.DataGridViewCellCancelEventArgs
  isExternal: false
  name: System.Windows.Forms.DataGridViewCellCancelEventArgs
- uid: System.ArgumentException
  isExternal: true
  name: System.ArgumentException
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Int32
  parent: System
  isExternal: true
  name: Int32
  nameWithType: Int32
  fullName: System.Int32
- uid: System.Windows.Forms.DataGridViewDataErrorContexts
  parent: System.Windows.Forms
  isExternal: false
  name: DataGridViewDataErrorContexts
  nameWithType: DataGridViewDataErrorContexts
  fullName: System.Windows.Forms.DataGridViewDataErrorContexts
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Context
  nameWithType: DataGridViewDataErrorEventArgs.Context
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Exception
  nameWithType: DataGridViewDataErrorEventArgs.Exception
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: ThrowException
  nameWithType: DataGridViewDataErrorEventArgs.ThrowException
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
- uid: System.Boolean
  parent: System
  isExternal: true
  name: Boolean
  nameWithType: Boolean
  fullName: System.Boolean
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: DataGridViewDataErrorEventArgs
  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Context
  nameWithType: DataGridViewDataErrorEventArgs.Context
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Exception
  nameWithType: DataGridViewDataErrorEventArgs.Exception
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: ThrowException
  nameWithType: DataGridViewDataErrorEventArgs.ThrowException
